contracts/liquidity_hub/pool-network/terraswap_pair/src/commands.rs

```diff
@@ -465,13 +466,28 @@ pub fn swap(
 
     #[cfg(feature = "osmosis")]
     if !swap_computation.osmosis_fee_amount.is_zero() {
-        let osmosis_fee_asset = Asset {
-            info: ask_pool.info.clone(),
-            amount: swap_computation.osmosis_fee_amount,
+        // send osmosis fee to the Community Pool
+        let denom = match ask_pool.info.clone() {
+            AssetInfo::Token { .. } => return Err(StdError::generic_err("Not supported").into()),
+            AssetInfo::NativeToken { denom } => denom,
         };
 
-        // send osmosis fee to the osmosis fee collector
-        messages.push(osmosis_fee_asset.into_msg(config.osmosis_fee_collector_addr)?);
+        //https://docs.cosmos.network/v0.45/core/proto-docs.html#cosmos.distribution.v1beta1.MsgFundCommunityPool
+        let community_pool_msg = CosmosMsg::Stargate {
+            type_url: "/cosmos.distribution.v1beta1.MsgFundCommunityPool".to_string(),
+            value: Anybuf::new()
+                .append_repeated_message(
+                    1,
+                    &[&Anybuf::new()
+                        .append_string(1, denom)
+                        .append_string(2, swap_computation.osmosis_fee_amount.to_string())],
+                )
+                .append_string(2, env.contract.address.to_string())
+                .into_vec()
+                .into(),
+        };
+
+        messages.push(community_pool_msg);
```

Comment:
This is kinda the "right way" to do it like you said. You could indeed send the tokens to the community pool address, but that requires you to set it up via the instantiate message. This way you don't need to, since the community pool address is fetched from the SDK when sent the said message.